pulsar/consumer_multitopic.go

```diff
@@ -45,54 +45,32 @@ func newMultiTopicConsumer(client *client, options ConsumerOptions, topics []str
 	mtc := &multiTopicConsumer{
 		options:   options,
 		messageCh: messageCh,
-		consumers: make(map[string]*consumer, len(topics)),
+		consumers: make(map[string]Consumer, len(topics)),
 		closeCh:   make(chan struct{}),
 		log:       &log.Entry{},
 	}
 
-	type ConsumerError struct {
-		err      error
-		topic    string
-		consumer *consumer
-	}
-
-	var wg sync.WaitGroup
-	wg.Add(len(topics))
-	ch := make(chan ConsumerError, len(topics))
-	for i := range topics {
-		go func(t string) {
-			defer wg.Done()
-			c, err := internalTopicSubscribe(client, options, t, messageCh)
-			ch <- ConsumerError{
-				err:      err,
-				topic:    t,
-				consumer: c,
-			}
-		}(topics[i])
-	}
-
-	go func() {
-		wg.Wait()
-		close(ch)
-	}()
-
 	var errs error
-	for ce := range ch {
+	consumers := make(map[string]Consumer, len(topics))
```

Comment:
Why do we need temporary consumers value here?

Reply:
That's a good question. The idea was to only set the consumers when everything succeeded but I don't think it matters since we return nil on failure. I'll remove it.